src/cplus_plugin/gui/component_item_model.py

```diff
@@ -127,10 +135,10 @@ def clone(self) -> "ModelComponentItemType":
 class LayerComponentItem(ModelComponentItem):
     """Base class view item for layer-based component items."""
 
-    def __init__(self, model_component: LayerModelComponent):
+    def __init__(self, model_component: LayerModelComponent, checkable: bool = False):
```

Comment:
@osundwajeff see earlier comment on having the checkable state defined in the `QStandardItemModel` sbclasses rather than `QStandardItem`.